tests/utils/verification/volumeproperties.go

```diff
@@ -90,17 +95,54 @@ func GetVolumePropertiesDockerCli(volName string, hostname string) string {
 	return op
 }
 
+// VerifyAttachedStatus - verify volume is attached and name of the VM attached
+// is consistent on both docker host and ESX
+func VerifyAttachedStatus(name, hostName, esxName string) bool {
+	log.Printf("Confirming attached status for volume [%s]\n", name)
+
+	vmAttachedHost := GetVMAttachedToVolUsingDockerCli(name, hostName)
+	vmAttachedESX := GetVMAttachedToVolUsingAdminCli(name, esxName)
+
+	return (vmAttachedHost == vmAttachedESX)
+}
+
+//GetVolumeStatusHost - get the volume status on a given host
+func GetVolumeStatusHost(name, hostName string) string {
+	cmd := dockercli.InspectVolume + " --format '{{index .Status.status}}' " + name
+	op := ExecCmd(hostName, cmd)
+	return strings.TrimSpace(op)
+}
+
+// VerifyDetachedStatus - check if the status gets detached within the timeout
+func VerifyDetachedStatus(name, hostName, esxName string) bool {
+	log.Printf("Confirming detached status for volume [%s]\n", name)
+	for attempt := 0; attempt < pollTimeout; attempt++ {
+		time.Sleep(1 * time.Second)
+		status := GetVolumeStatusHost(name, hostName)
+		if status != admincli.DetachedStatus {
+			continue
+		}
+		// this api returnes "detached" in when volume is detached
+		status = GetVMAttachedToVolUsingAdminCli(name, esxName)
+		if status == admincli.DetachedStatus {
+			return true
+		}
+	}
+	log.Printf("Timed out to poll status\n")
```

Comment:
would suggest to use `log.Warn` or `log.Error` instead of `log.Printf` that helps during debugging test failure